ingest/src/main/java/com/microsoft/azure/kusto/ingest/resources/RankedStorageAccount.java

```diff
@@ -82,22 +95,25 @@ public double getRank() {
         // For each bucket, calculate the success rate ( success / total ) and multiply it by the bucket weight.
         // The older the bucket, the less weight it has. For example, if there are 3 buckets, the oldest bucket will have
         // a weight of 1, the middle bucket will have a weight of 2 and the newest bucket will have a weight of 3.
-
-        for (Bucket bucket : buckets) {
-            if (bucket.totalCount == 0) {
+        try {
+            bucketsLock.readLock().lock();
+            for (Bucket bucket : buckets) {
```

Comment:
@AsafMah was right :-)
So since the modification happens in other code, we have to use `locks`.